python/pip_install/extract_wheels/lib/BUILD

```diff
@@ -70,18 +70,22 @@ py_test(
     ],
 )
 
-py_test(
-    name = "whl_filegroup_test",
-    size = "small",
-    srcs = [
-        "whl_filegroup_test.py",
-    ],
-    data = ["//examples/wheel:minimal_with_py_package"],
-    tags = ["unit"],
-    deps = [
-        ":lib",
-    ],
-)
+# The `whl_filegroup_test` changes directories at runtime which impacts the
+# behavior of all other tests. To avoid this issue, each test is broken out
+# into unique targets.
+[
+    py_test(
+        name = "whl_filegroup_{}_test".format(incremental),
+        size = "small",
+        srcs = ["whl_filegroup_test.py"],
+        data = ["//examples/wheel:minimal_with_py_package"],
+        env = {"INCREMENTAL": incremental},
+        main = "whl_filegroup_test.py",
+        tags = ["unit"],
+        deps = [":lib"],
+    )
+    for incremental in ("incremental", "non_incremental")
```

Comment:
or, this seems like a manual sharding scheme, we could just set a shard_count to get them isolated

Reply:
refactored this to not be necessary.